DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UIActivityIndicatorRecorder.swift

```diff
@@ -0,0 +1,71 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+#if os(iOS)
+import UIKit
+
+internal struct UIActivityIndicatorRecorder: NodeRecorder {
+    let identifier = UUID()
+
+    func semantics(of view: UIView, with attributes: ViewAttributes, in context: ViewTreeRecordingContext) -> NodeSemantics? {
+        guard let activityIndicator = view as? UIActivityIndicatorView else {
+            return nil
+        }
+
+        guard attributes.isVisible else {
+            return InvisibleElement.constant
+        }
+
+        guard activityIndicator.isAnimating || !activityIndicator.hidesWhenStopped else {
+            return InvisibleElement.constant
+        }
+
+        let builder = UIActivityIndicatorWireframesBuilder(
+            attributes: attributes,
+            wireframeID: context.ids.nodeID(view: activityIndicator, nodeRecorder: self),
+            backgroundColor: activityIndicator.backgroundColor?.cgColor
+        )
+
+        let node = Node(viewAttributes: attributes, wireframesBuilder: builder)
+        let allNodes = [node] + recordSubtree(of: activityIndicator, in: context)
+        return SpecificElement(subtreeStrategy: .ignore, nodes: allNodes)
+    }
+
+    private func recordSubtree(of activityIndicator: UIActivityIndicatorView, in context: ViewTreeRecordingContext) -> [Node] {
+        let subtreeViewRecorder = ViewTreeRecorder(
+            nodeRecorders: [
+                UIImageViewRecorder(
+                    shouldRecordImagePredicate: { imageView in
+                        return imageView.image == nil ? false : true
+                    }
```

Comment:
/nit
```suggestion
                    shouldRecordImagePredicate: { $0.image != nil }
```
/question Are we forcing recording of the image because it is not considered as context image?

Reply:
Yes, exactly!